.changeset/good-elephants-knock.md

```diff
@@ -0,0 +1,5 @@
+---
+'@directus/env': patch
+---
+
+Fixed behavior of config variable casting to retain types for boolean values
```

Comment:
Nice patch! Should we change `boolean` to `typed`?

```suggestion
Fixed behavior of config variable casting to retain types for typed values
```

Or the following perhaps

```suggestion
Retained config variables types for known value types
```